Sources/SourceKitLSP/Swift/OpenInterface.swift

```diff
@@ -10,129 +10,51 @@
-        try await sendSourcekitdRequest(closeDocumentSourcekitdRequest(uri: interfaceDocURI), fileContents: nil)
-      }
-      return result
-    }
-  }
-
-  /// Open the Swift interface for a module.
-  ///
-  /// - Parameters:
-  ///   - document: The document whose compiler arguments should be used to generate the interface.
-  ///   - moduleName: The module to generate an index for.
-  ///   - groupName: The module group name.
-  ///   - interfaceURI: The file where the generated interface should be written.
-  ///
-  /// - Important: This opens a document with name `interfaceURI.pseudoPath` in sourcekitd. The caller is responsible
-  ///   for ensuring that the document will eventually get closed in sourcekitd again.
-  private func generatedInterfaceInfo(
-    document: DocumentURI,
-    moduleName: String,
-    groupName: String?,
-    interfaceURI: DocumentURI
-  ) async throws -> GeneratedInterfaceInfo {
-    let keys = self.keys
-    let skreq = sourcekitd.dictionary([
-      keys.request: requests.editorOpenInterface,
-      keys.moduleName: moduleName,
-      keys.groupName: groupName,
-      keys.name: interfaceURI.pseudoPath,
-      keys.synthesizedExtension: 1,
-      keys.compilerArgs: await self.buildSettings(for: document, fallbackAfterTimeout: false)?.compilerArgs
-        as [SKDRequestValue]?,
-    ])
-
-    let dict = try await sendSourcekitdRequest(skreq, fileContents: nil)
-    return GeneratedInterfaceInfo(contents: dict[keys.sourceText] ?? "")
-  }
-
-  private func generatedInterfaceDetails(
-    uri: DocumentURI,
-    snapshot: DocumentSnapshot,
-    symbol: String?
-  ) async -> GeneratedInterfaceDetails {
-    do {
-      guard let symbol = symbol else {
-        return GeneratedInterfaceDetails(uri: uri, position: nil)
+        nil
       }
-      let keys = self.keys
-      let skreq = sourcekitd.dictionary([
-        keys.request: requests.editorFindUSR,
-        keys.sourceFile: uri.sourcekitdSourceFile,
-        keys.primaryFile: uri.primaryFile?.pseudoPath,
-        keys.usr: symbol,
-      ])
 
-      let dict = try await sendSourcekitdRequest(skreq, fileContents: snapshot.text)
-      if let offset: Int = dict[keys.offset] {
-        return GeneratedInterfaceDetails(uri: uri, position: snapshot.positionOf(utf8Offset: offset))
-      } else {
-        return GeneratedInterfaceDetails(uri: uri, position: nil)
-      }
-    } catch {
-      return GeneratedInterfaceDetails(uri: uri, position: nil)
+    if case .dictionary(let experimentalCapabilities) = self.capabilityRegistry.clientCapabilities.experimental,
+      case .bool(true) = experimentalCapabilities["workspace/getReferenceDocument"]
+    {
+      return GeneratedInterfaceDetails(uri: try urlData.uri, position: position)
     }
+    let interfaceFilePath = self.generatedInterfacesPath.appendingPathComponent(urlData.displayName)
+    try await generatedInterfaceManager.snapshot(of: urlData).text.write(
```

Comment:
@ahoppen this looks like we no longer cache generated .swiftinterface files unless the LSP client says it supports `workspace/getReferenceDocument`, i.e. only VS Code users using the official extension.

Is that intentional? I would have thought we'd still want to cache regardless. Switching from Swift 6.1 to 6.2 has regressed go-to-def time on AttributedString from 1.0 to 4.6 seconds in my testing, because Foundation.swiftinterface is no longer cached.

Reply:
I've filed an issue: https://github.com/swiftlang/sourcekit-lsp/issues/2369